src/providers/defaults.ts

```diff
@@ -60,114 +36,135 @@ export async function setDefaultEmbeddingProviders(provider: ApiProvider) {
   defaultEmbeddingProvider = provider;
 }
 
+/**
+ * Helper to check if a key is set in environment or env overrides
+ * Exported for testing
+ */
+export function isKeySet(key: keyof EnvOverrides, env?: EnvOverrides): boolean {
+  return !!getEnvString(key) || !!env?.[key];
+}
+
+/**
+ * Simplified provider configuration with credential checking
+ */
+interface ProviderEntry {
+  /** Provider name for logging */
+  name: string;
+
+  /** Provider configuration function */
+  config: ProviderConfiguration;
+
+  /** Check if this provider's credentials are available */
+  hasCredentials(env?: EnvOverrides): boolean | Promise<boolean>;
+}
+
+// Exported for testing
+export const PROVIDERS: ProviderEntry[] = [
```

Comment:
add a note that order matters.  so you don't come back in 1 year and make it alphabetical